main.rb

```diff
@@ -0,0 +1,140 @@
+# rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      if is_a?(Array)
+        yield(self[index])
+      elsif is_a?(Hash)
+        yield(keys[index], self[keys[index]])
+      elsif is_a?(Range)
+        yield(to_a[index])
+      else
+        yield
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    while index < size
+      if is_a?(Array)
+        yield(self[index], index)
+      elsif is_a?(Hash)
+        yield(keys[index], self[keys[index]])
+      elsif is_a?(Range)
+        yield(to_a[index], index)
+      else
+        yield
+      end
+      index += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array) || is_a?(Range)
+      selected_array = []
+      my_each { |item| selected_array.push(item) if yield(item) }
+      selected_array
+    else
+      selected_hash = {}
+      my_each { |key, value| selected_hash[key] = value if yield(key, value) }
+      selected_hash
+    end
+  end
+
+  def my_all?(my_arg = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false || yield(item).nil? }
+    else
+      my_each { |item| return false if item == false || item.nil? }
+    end
+    if my_arg.class == Regexp
+      my_each { |item| item =~ my_arg }
+    elsif my_arg.class == Class
+      my_each { |item| item.is_a? my_arg }
+    else
+      my_each { |item| item == my_arg }
+    end
+    true
+  end
+
+  def my_any?(my_arg = nil)
+    if block_given?
+      if is_a?(Array) || is_a?(Range)
+        my_each { |item| return true if yield(item) == true }
+      else
+        my_each { |key, item| return true if yield(key, item) == true }
+      end
+    end
+    if my_arg == Regexp
+      my_each { |item| return true if item.to_s.match(my_arg) }
+    elsif my_arg.is_a? Class
+      my_each { |item| return true if item.is_a? my_arg }
+    else
+      my_each { |item| return true if item == true }
+    end
+    false
+  end
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
['saheed', 'oladele', 'suretrust'].my_any?(/d/) != ['saheed', 'oladele', 'suretrust'].any?(/d/)

['cat', 'dog'].my_any?('cat') != ['cat', 'dog'].any?('cat')
```

Reply:
Fixed.